### PR TITLE
[#3360] Create native executable command line client

### PR DESCRIPTION
**Pull request by sophokles73**
Addresses #3360 

The build jobs have been changed to consistently use the
eclipse/hono-builder container image for building Hono. This image
contains specific versions of Maven, the GraalVM SDK with the
native-image compiler plugin and the UPX tool for compressing binary
artifacts.

The release build job has been extended with an additional invocation
of mvn clean install used to build the native executable of the CLI.

The native executable needs to be built in a separate run because in the
first build the JAR file artifacts are being signed by the Eclipse JAR
signer service which results in signature errors during the native
executable build.

### CODE REVIEW

File: cli/pom.xml [hunk 1]
```diff
@@ -30,6 +30,19 @@
   <description>Quarkus based Hono Command line Interface</description>
```
Comment (calohmn) [inline cli/pom.xml:30]: typo in license header: Li2cense

**Comment by sophokles73** [comment]
@calohmn I have pushed changes

**Comment by calohmn** [comment]
@sophokles73 There are these warnings during the build. I guess they can be ignored?
```
[INFO] --- quarkus-maven-plugin:2.11.2.Final:build (default) @ hono-cli ---
[WARNING] [io.quarkus.deployment.steps.ReflectiveHierarchyStep] Unable to properly register the hierarchy of the following classes for reflection as they are not in the Jandex index:
	- io.micrometer.core.instrument.MeterRegistry (source: PicocliNativeImageProcessor)
	- io.micrometer.core.instrument.binder.kafka.KafkaClientMetrics (source: PicocliNativeImageProcessor)
	- io.opentracing.Span (source: PicocliNativeImageProcessor)
	- io.opentracing.Tracer (source: PicocliNativeImageProcessor)
	- io.vertx.core.AsyncResult (source: PicocliNativeImageProcessor)
	- io.vertx.core.Context (source: PicocliNativeImageProcessor)
	- io.vertx.core.Future (source: PicocliNativeImageProcessor)
	- io.vertx.core.Handler (source: PicocliNativeImageProcessor)
	- io.vertx.core.Promise (source: PicocliNativeImageProcessor)
	- io.vertx.core.Vertx (source: PicocliNativeImageProcessor)
	- io.vertx.core.buffer.Buffer (source: PicocliNativeImageProcessor)
	- io.vertx.core.net.KeyCertOptions (source: PicocliNativeImageProcessor)
	- io.vertx.core.net.TrustOptions (source: PicocliNativeImageProcessor)
	- io.vertx.kafka.client.common.TopicPartition (source: PicocliNativeImageProcessor)
	- io.vertx.kafka.client.consumer.KafkaConsumer (source: PicocliNativeImageProcessor)
	- io.vertx.kafka.client.consumer.KafkaConsumerRecord (source: PicocliNativeImageProcessor)
	- io.vertx.kafka.client.producer.KafkaProducer (source: PicocliNativeImageProcessor)
	- io.vertx.proton.ProtonClient (source: PicocliNativeImageProcessor)
	- io.vertx.proton.ProtonClientOptions (source: PicocliNativeImageProcessor)
	- io.vertx.proton.ProtonConnection (source: PicocliNativeImageProcessor)
	- io.vertx.proton.ProtonDelivery (source: PicocliNativeImageProcessor)
	- io.vertx.proton.ProtonReceiver (source: PicocliNativeImageProcessor)
	- io.vertx.proton.ProtonSender (source: PicocliNativeImageProcessor)
	- io.vertx.proton.ProtonSession (source: PicocliNativeImageProcessor)
	- javax.enterprise.event.Event (source: PicocliNativeImageProcessor)
	- org.apache.qpid.proton.amqp.Symbol (source: PicocliNativeImageProcessor)
	- org.apache.qpid.proton.message.Message (source: PicocliNativeImageProcessor)
	- org.eclipse.microprofile.health.HealthCheckResponse (source: PicocliNativeImageProcessor)
	- org.eclipse.microprofile.health.HealthCheckResponse$Status (source: PicocliNativeImageProcessor)
Consider adding them to the index either by creating a Jandex index for your dependency via the Maven plugin, an empty META-INF/beans.xml or quarkus.index-dependency properties.
```

**Comment by sophokles73** [comment]
At least they seem to have no (negative) impact on the produced executable ....

**Pull request merged**